File: hermes/drones/drone.py
#/usr/bin/python3

from abc import abstractmethod

class Drone:

    def __init__(self):
        pass

    @abstractmethod
    def connect(self):
        pass

    @abstractmethod
    def disconnect(self):
        pass

    @abstractmethod
    def takeOff(self):
        pass

    @abstractmethod
    def land(self):
        pass

    @abstractmethod
    def setHome(self, lat, lng):
        pass

    @abstractmethod
    def moveTo(self, lat, lng, alt):
        pass

    @abstractmethod
    def moveBy(self, x, y, z):
        pass

    @abstractmethod
    def rotateBy(self, theta):
        pass

    @abstractmethod
    def rotateTo(self, theta):
        pass

    @abstractmethod
    def gimbalSetPose(self, yaw_theta, pitch_theta, roll_theta):
        pass

    @abstractmethod
    def takePhoto(self):
        pass

    @abstractmethod
    def getVideoFrame(self):
        pass

    @abstractmethod
    def getStatus(self):
        pass

    @abstractmethod
    def cancel(self):
        pass

    

File: hermes/src/edu/cmu/cs/dronebrain/interfaces/Drone.java
package edu.cmu.cs.dronebrain.interfaces;

public interface Drone {
    void connect(String ip);
    void disconnect();
    void takeOff();
    void land();
    void setHome(Double lat, Double lng);
    void moveTo(Double lat, Double lng, Double alt);
    void moveBy(Integer x, Integer y, Integer z); //x,y,z in meters
    void startStreaming(Integer sample_rate);
    void rotateBy(Double theta);
    void rotateTo(Double theta);
    void setGimbalPose(Double yaw_theta, Double pitch_theta, Double roll_theta);
    void takePhoto();
    void getVideoFrame();
    void getStatus();
    void cancel();
}

